dataprofiler/profilers/profile_builder.py

```diff
@@ -1857,6 +1864,47 @@ def _merge_correlation_helper(corr_mat1, mean1, std1, n1,
 
         return corr_mat
 
+    def _update_chi2(self):
+        n_cols = len(self._profile)
+        cat_cols = {}
+        cat_col_idxs = []
+
+        # Get categorical columns
+        for i in range(n_cols):
```

Comment:
I wonder if it would be better to just loop through the columns instead of collecting them each time and then looping through the collected categories.

something like, define an internal function which gets a categorical column or returns None
```python
def get_cateogrical_profile_if_match(col):
    data_stats_compiler = compiler.profiles["data_stats_profile"]
     profiler = data_stats_compiler._profiles["category"]
     return profiler if profiler.is_match else None
```
then you could loop through the columns
```
for i in range(n_cols):
    col1_cat = get_cateogrical_profile_if_match(self._profile[i])
    if col1_cat is None:
        continue
    for j in range(i, n_cols):
        col2_cat = get_cateogrical_profile_if_match(self._profile[j])
        .
        .
        .
```
thoughts? 
it minimizes the code at least to a specific area, but not sure that's necessary.